src/nnbench/reporter/file.py

```diff
@@ -0,0 +1,205 @@
+parsers = {"json": JsonParser, "yaml": YamlParser}
+
+
+def parse_records(records: str, file_type: str) -> Any:
+    """Parses records based on the specified file type.
+
+    This function retrieves and calls the registered parser for
+    the given file type.
+
+    Args:
+        `records:` A list or iterator of record strings.
+        `file_type:` The file type (string).
+
+    Returns:
+        A list of parsed records.
+    """
+
+    parser = Parser.get_parser(file_type)
+    if parser is None:
+        raise ValueError(f"Unsupported file type: {file_type}")
+
+    return parser().parse_file(records)
+
+
+def append_record_to_records(parsed_records: Any, record: BenchmarkRecord, file_type: str) -> str:
+    """Appends a record to the list based on the file type.
+
+    This function first parses the record using the appropriate parser
+    and then appends it to the `parsed_records`.
+
+    Args:
+        `records:` A list of parsed records.
+        `record:` The record to append.
+        `file_type:` The file type (string).
+    """
+
+    parser = Parser.get_parser(file_type)
+    if parser is None:
+        raise ValueError(f"Unsupported file type: {file_type}")
+
+    return parser().write_records(parsed_records, record)
+
+
+class FileReporter(BenchmarkReporter):
+    def __init__(self, dir: str):
+        self.dir = dir
+        if not os.path.exists(dir):
+            self.initialize()
+
+    def initialize(self) -> None:
+        try:
+            os.makedirs(self.dir, exist_ok=True)
+        except OSError as e:
+            self.finalize()
+            raise ValueError(f"Could not create directory: {self.dir}") from e
+
+    def read(self, file_name: str) -> BenchmarkRecord:
+        if not self.dir:
+            raise BaseException("Directory is not initialized")
+        file_path = os.path.join(self.dir, file_name)
+        file_type = file_name.split(".")[1]
+        try:
+            with open(file_path) as file:
+                data = file.read()
+                parsed_data = parse_records(data, file_type)
+                return parsed_data
+        except FileNotFoundError:
+            raise ValueError(f"Could not read the file: {file_path}")
+
+    def write(self, record: BenchmarkRecord, file_name: str) -> None:
+        if not self.dir:
+            raise BaseException("Directory is not initialized")
+
+        file_path = os.path.join(self.dir, file_name)
+        if not os.path.exists(file_path):  # Create the file
+            with open(file_path, "w") as file:
+                file.write("")
+        try:
+            parsed_records = self.read(file_name)
+            file_type = file_name.split(".")[1]
+            new_records = append_record_to_records(parsed_records, record, file_type)
+            with open(file_path, "w") as file:
+                file.write(new_records)
+        except FileNotFoundError:
+            raise ValueError(f"Could not read the file: {file_path}")
```

Comment:
Same here, just load the serializer from the dict and call it on the opened file.